nil pointer dereference on CreateVM

**What happened:**
Create vm fails with nil pointer dereference.

**What did you expect to happen:**
It does not panic with `nil pointer dereference`. If I did something wrong, the system should tell me what.

**How to reproduce it:**
1. remove initrd from the example payload in the repo
2. set rootfs to `is_root:true`
3. called `./hack/scripts/send.sh -m CreateMicroVM`

**Anything else you would like to add:**
```
INFO[0001] checking state of microvm                     controller=microvm service=firecracker_microvm vmid=ns1/mvm1
DEBU[0001] execute step                                  controller=microvm execution_id=01FK0TBQB7NB4MS1AX1XY10N3Y plan_name=microvm_create step=microvm_create
DEBU[0001] creating microvm                              controller=microvm step=microvm_create vmid=ns1/mvm1
DEBU[0001] creating microvm                              controller=microvm service=firecracker_microvm vmid=ns1/mvm1
DEBU[0002] PUT /machine-config HTTP/1.1
Host: localhost
User-Agent: Go-http-client/1.1
Content-Length: 56
Accept: application/json
Content-Type: application/json
Accept-Encoding: gzip

{"ht_enabled":false,"mem_size_mib":2048,"vcpu_count":2}
  controller=microvm service=firecracker_microvm vmid=ns1/mvm1
DEBU[0002] HTTP/1.1 204
Content-Length: 0
Connection: keep-alive
Server: Firecracker API

  controller=microvm service=firecracker_microvm vmid=ns1/mvm1
DEBU[0002] PUT /drives/root HTTP/1.1
Host: localhost
User-Agent: Go-http-client/1.1
Content-Length: 114
Accept: application/json
Content-Type: application/json
Accept-Encoding: gzip

{"drive_id":"root","is_read_only":false,"is_root_device":true,"path_on_host":"/dev/mapper/dev-thinpool-snap-104"}
  controller=microvm service=firecracker_microvm vmid=ns1/mvm1
DEBU[0002] HTTP/1.1 204
Content-Length: 0
Connection: keep-alive
Server: Firecracker API

  controller=microvm service=firecracker_microvm vmid=ns1/mvm1
DEBU[0002] PUT /network-interfaces/eth0 HTTP/1.1
Host: localhost
User-Agent: Go-http-client/1.1
Content-Length: 110
Accept: application/json
Content-Type: application/json
Accept-Encoding: gzip

{"allow_mmds_requests":true,"guest_mac":"AA:FF:00:00:00:01","host_dev_name":"ns1_mvm1_tap","iface_id":"eth0"}
  controller=microvm service=firecracker_microvm vmid=ns1/mvm1
DEBU[0002] HTTP/1.1 204
Content-Length: 0
Connection: keep-alive
Server: Firecracker API

  controller=microvm service=firecracker_microvm vmid=ns1/mvm1
DEBU[0002] PUT /network-interfaces/eth1 HTTP/1.1
Host: localhost
User-Agent: Go-http-client/1.1
Content-Length: 82
Accept: application/json
Content-Type: application/json
Accept-Encoding: gzip

{"guest_mac":"4E:1D:39:56:BC:6D","host_dev_name":"/dev/tap350","iface_id":"eth1"}
  controller=microvm service=firecracker_microvm vmid=ns1/mvm1
DEBU[0002] HTTP/1.1 204
Content-Length: 0
Connection: keep-alive
Server: Firecracker API

  controller=microvm service=firecracker_microvm vmid=ns1/mvm1
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x12cb697]

goroutine 81 [running]:
github.com/weaveworks/flintlock/infrastructure/firecracker.ApplyConfig({0x16f0540, 0xc000616420}, 0xc00026e8c0, 0x1f1c050)
        /home/efertone/Development/github.com/weaveworks/flintlock/infrastructure/firecracker/config.go:163 +0x217
github.com/weaveworks/flintlock/infrastructure/firecracker.(*fcProvider).Create(0xc000290de0, {0x16f0540, 0xc000616420}, 0xc000c99100)
        /home/efertone/Development/github.com/weaveworks/flintlock/infrastructure/firecracker/create.go:77 +0xb15
github.com/weaveworks/flintlock/core/steps/microvm.(*createStep).Do(0xc000889368, {0x16f0540, 0xc000616420})
        /home/efertone/Development/github.com/weaveworks/flintlock/core/steps/microvm/create.go:53 +0x203
github.com/weaveworks/flintlock/pkg/planner.(*actuatorImpl).react(0xc000616390, {0x16f0540, 0xc000616420}, {0xc000c9be00, 0x6, 0xc000d27588}, 0xc000d275d8)
        /home/efertone/Development/github.com/weaveworks/flintlock/pkg/planner/actuator.go:96 +0x2be
github.com/weaveworks/flintlock/pkg/planner.(*actuatorImpl).executePlan(0xc0006cc690, {0x16f0540, 0xc000616420}, {0x16de268, 0xc000616390}, 0xc000d27670)
        /home/efertone/Development/github.com/weaveworks/flintlock/pkg/planner/actuator.go:68 +0xbd
github.com/weaveworks/flintlock/pkg/planner.(*actuatorImpl).Execute(0x16f0540, {0x16f0540, 0xc000616420}, {0x16de268, 0xc000616390}, {0xc00034d000, 0x1a})
        /home/efertone/Development/github.com/weaveworks/flintlock/pkg/planner/actuator.go:37 +0x294
github.com/weaveworks/flintlock/core/application.(*app).reconcile(0xc0002850e0, {0x16f0540, 0xc00032e600}, 0xc000c99100, 0x4)
        /home/efertone/Development/github.com/weaveworks/flintlock/core/application/reconcile.go:112 +0x2fd
github.com/weaveworks/flintlock/core/application.(*app).ReconcileMicroVM(0xc0002850e0, {0x16f0540, 0xc00032e600}, {0xc0009df374, 0x4}, {0xc0009df370, 0x3})
        /home/efertone/Development/github.com/weaveworks/flintlock/core/application/reconcile.go:25 +0x27c
github.com/weaveworks/flintlock/infrastructure/controllers.(*MicroVMController).processQueueItem(0xc00032e540, {0x16f0540, 0xc00032e600})
        /home/efertone/Development/github.com/weaveworks/flintlock/infrastructure/controllers/microvm_controller.go:128 +0x1a3
github.com/weaveworks/flintlock/infrastructure/controllers.(*MicroVMController).Run.func3()
        /home/efertone/Development/github.com/weaveworks/flintlock/infrastructure/controllers/microvm_controller.go:65 +0x85
created by github.com/weaveworks/flintlock/infrastructure/controllers.(*MicroVMController).Run
        /home/efertone/Development/github.com/weaveworks/flintlock/infrastructure/controllers/microvm_controller.go:63 +0x475
exit status 2
```

**Environment:**

- flintlock version: Commit: `3217c47f5bb2bef558a8c66cf5dbce4ae044cda5` (`main` right now)

```
NAME="Ubuntu"
VERSION="20.04.3 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.3 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```

--- a/infrastructure/firecracker/config.go
+++ b/infrastructure/firecracker/config.go
@@ -157,14 +157,26 @@ func ApplyConfig(ctx context.Context, cfg *VmmConfig, client *firecracker.Client
 			return fmt.Errorf("putting %s network configuration: %w", guestIfaceName, err)
 		}
 	}
-	_, err = client.PutGuestBootSource(ctx, &fcmodels.BootSource{
+
+	bootSource := fcmodels.BootSource{
+		BootArgs:        "",
+		InitrdPath:      "",
 		KernelImagePath: &cfg.BootSource.KernelImagePage,
-		BootArgs:        *cfg.BootSource.BootArgs,
-		InitrdPath:      *cfg.BootSource.InitrdPath,
-	})
+	}
+
+	if cfg.BootSource.BootArgs != nil {
+		bootSource.BootArgs = *cfg.BootSource.BootArgs
+	}
+
+	if cfg.BootSource.InitrdPath != nil {
+		bootSource.InitrdPath = *cfg.BootSource.InitrdPath
+	}
+
+	_, err = client.PutGuestBootSource(ctx, &bootSource)
 	if err != nil {
 		return fmt.Errorf("failed to put machine bootsource: %w", err)
 	}
+
 	if cfg.Logger != nil {
 		_, err = client.PutLogger(ctx, &fcmodels.Logger{
 			LogPath:       &cfg.Logger.LogPath,

